flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -52,13 +56,15 @@
 	 * @param restartBackoffTimeStrategy helps to decide whether to restart failed tasks and the restarting delay
 	 */
 	public ExecutionFailureHandler(
-		FailoverTopology<?, ?> failoverTopology,
-		FailoverStrategy failoverStrategy,
-		RestartBackoffTimeStrategy restartBackoffTimeStrategy) {
+			final FailoverTopology<?, ?> failoverTopology,
+			final FailoverStrategy failoverStrategy,
+			final RestartBackoffTimeStrategy restartBackoffTimeStrategy,
+			final Counter numberOfRestartsCounter) {
```

Comment:
I would be in favor not to pass another dependency via the constructor. It would be possible to poll the counter from outside the `ExecutionFailureHandler`, if the metric was a `Gauge`.

Reply:
See my other comment. Gauge is a better choice if we can have the conclusion to not add a `numberOfRestartsPerSecond` meter.

Reply:
Replied.

Reply:
done.